src/compatTest/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginFunctionalTest.groovy

```diff
@@ -405,26 +328,99 @@ sourceSets {
                   |scalafixTest - Runs Scalafix on Scala sources in 'test'""".stripMargin())
     }
 
-    BuildResult runGradleTask(String task, List<String> arguments) {
+    @Unroll
+    def 'it should run semantic and syntactic rules in projects using Scala #scalaVersion'() {
+        given:
+        def projectDir = createScalaProject("", scalaVersion)
+        createScalafixConfig(projectDir, '''
+rules = [ 
+  RemoveUnused
+  NoAutoTupling
+  DisableSyntax
+  LeakingImplicitClassVal
+  NoValInForComprehension
+  ProcedureSyntax 
+ ]
+ 
+DisableSyntax.noVars = true
+''')
+        createSourceFile(projectDir, 'trait Animal')
+        createSourceFile(projectDir, 'object Dog extends Animal')
+        createSourceFile(projectDir, 'object Cat extends Animal')
+        createSourceFile(projectDir, 'class Bird extends Animal')
+
+        when:
+        def buildResult = runGradleTask(projectDir, 'scalafix', ['--stacktrace'])
+
+        then:
+        buildResult.output.contains('BUILD SUCCESSFUL')
+
+        where:
+        scalaVersion ||_
+        '2.11.8'     ||_
+        '2.11.9'     ||_
+        '2.11.10'    ||_
+        '2.11.11'    ||_
+        '2.11.12'    ||_
+        '2.12.4'     ||_
+        '2.12.5'     ||_
+        '2.12.6'     ||_
+        '2.12.7'     ||_
+        '2.12.8'     ||_
+        '2.12.9'     ||_
+        '2.12.10'    ||_
+        '2.13.0'     ||_
+        '2.13.1'     ||_
+    }
```

Comment:
these are the versions of Scala that seem to work using the `scalametaVersion` specified in `scalafix-interfaces.properties` file. Even though the SemanticDB compiler plugin used to be published to older versions of Scala 2.11.x and 2.12.x, their versions are a bit old and I don't think we should maintain any sort of mapping of what version of SemanticDB works with what version of Scala.

The only exception here is Scala 2.13. The `scala213` version in `scalafix-interfaces.properties` specifies version 2.13.0 of Scala. It turns out that there is a breaking change in `scala-compiler` from 2.13.0 to 2.13.1. So `semanticdb_2.13.0` does not work with Scala compiler 2.13.1, neither `semanticdb_2.13.1` works with Scala compiler 2.13.0:
- https://github.com/scalameta/scalameta/blob/master/semanticdb/scalac/library/src/main/scala-2.13.0-/scala/meta/internal/semanticdb/scalac/SemanticdbReporter.scala
- https://github.com/scalameta/scalameta/blob/master/semanticdb/scalac/library/src/main/scala-2.13.1+/scala/meta/internal/semanticdb/scalac/SemanticdbReporter.scala

As a workaround, I'm using whatever version of Scala the project where the plugin is applied is using. Let's see how this is going to play out with future versions of Scala.

BTW, the Scalafix docs say it only works with the latest version of each Scala 2.1x series:

![image](https://user-images.githubusercontent.com/1107367/75503690-455bca80-59ce-11ea-976f-5e633331ef83.png)

But in practice it seems to work with a larger range of versions of Scala. This PR leaves that possibility open to whoever wants to take the risk.